app/javascript/controllers/reloadable_controller.js

```diff
@@ -0,0 +1,55 @@
+import { Controller } from "@hotwired/stimulus"
+import { API } from '../http_api'
+import { miqFetch, addSearchParams } from '../http_api/fetch'
+
+export default class extends Controller {
+  static targets = ["content"]
+  static values = {url: String}
+
+  async reload() {
+    try {
+      window.miqSparkleOn()
+
+      // request the new content and get back a task id
+      const data = await miqFetch({
+        url: this.urlValue,
+        method: 'POST',
+        cookieAndCsrf: true
+      })
+
+      const task_id = data.task_id
+      if (!task_id) {
+        throw new Error('No task_id returned from server')
+      }
+
+      // poll for the task status
+      await API.wait_for_task(task_id)
```

Comment:
This is different from how the old way worked, and I'm wondering if we should switch to the old way. The old way doesn't use the API. Instead it hits the wait_for_task endpoint and uses the /report/wait_for_task endpoint. I think this might be better in the case someone doesn't have access to the /api/tasks endpoint, they _would_ have access to this /report/wait_for_task endpoint. I gotta play around with access. I might have to completely rework wait_for_task though, since it's a weird polling mechanism that relies on the client-side rjs to come back in.